include/odp/api/spec/crypto.h

```diff
@@ -127,6 +140,9 @@ typedef enum {
 	 */
 	ODP_CIPHER_ALG_ZUC_EEA3,
 
+	/** AES_CTR in EEA2 mode */
+	ODP_CIPHER_ALG_AES_CTR_EEA2,
```

Comment:
This should be aligned with current 128-EEA1 and 128-EEA3 definitions:
1) It would be better to have it defined between ODP_CIPHER_ALG_SNOW3G_UEA2 and ODP_CIPHER_ALG_ZUC_EEA3
2) CTR is not needed in the name as EEA2 spec define the AES mode: better name is  ODP_CIPHER_ALG_AES_EEA2
3) Mention 128-EEA2 in comment

ODP is maintaining the counter value, which includes five "bearer" bits. How application passes bearer to the  crypto operation / session ?

Reply:
Ok

Counter and bearer are passed as pre-formatted IV blocks. If we need higher level API, it can be created on top crypto API. Do you have any suggestions on how this high level API should look?

Reply:
API spec needs some specification about how bearer is passed / IV is used.

We can refer to 3GPP spec (33.401: chapter B.1.3 128-EEA2 ?) format. According to that CTR = COUNT[0..31] | BEARER[0..4] | DIRECTION | 26 bits zero | 64bit block counter. User would need to pass COUNT and BEARER for every operation through IV.

Other option would be to add COUNT/BEARER to crypto session params so that ODP would update/fill those automatically.

Reply:
@psavol does the following sound good to you:

```
For 3GPP algorithms users of crypto API should format IV blocks by hand according to the 3GPP specificatons using COUNTER/BEARER/DIRECTION.
```

Reply:
Maybe a bit more specific:

"3GPP algorithms require initialization vector to be formatted as specified in 3GPP standards (see e.g. TS 33.401). COUNT/BEARER/DIRECTION information needs to be packed into IV bits as standards specify." 

Then also specification if override or session IV may be used.
"128-EEA2 require IV to be set per operation (due to unique COUNT value)"

Did not check it, but may be e.g. ZUC can be used also with session IV, if the format does not have a counter.

Reply:
All 3gpp algorithms use `COUNT`/`BEARER`/`DIRECTION` (and some of them require random called `FRESH`.

Reply:
@psavol I ended up describing IV format for all 8 algorithms.